### PR TITLE
fix: revert #5458

### PR DESCRIPTION
**Pull request by cartant**
**Description:**

This PR adds a failing test and reverts #5458 - which removed the `isComplete` variable from `shareReplay`. With the reversion, the test passes. There was an existing test that attempted to test what's tested by the new test, but there was a subtle difference because of this problem: https://github.com/ReactiveX/rxjs/issues/5523

**Related issue (if exists):** #5458

### CODE REVIEW

**Comment by josepot** [comment]
FYI @cartant @benlesh #5550 also adds the test and it prevents a possible edge-case that I explained in [this comment](https://github.com/ReactiveX/rxjs/pull/5458#issuecomment-652544364).

File: spec/operators/shareReplay-spec.ts [hunk 1]
```diff
@@ -219,6 +219,21 @@ describe('shareReplay operator', () => {
     expectSubscriptions(source.subscriptions).toBe(sourceSubs);
   });
 
+  it('should not restart a synchronous source due to unsubscriptions if refCount is true when the source has completed', () => {
+    // The test above this one doesn't actually test completely synchronous
```
Comment (josepot) [inline spec/operators/shareReplay-spec.ts:223]: I'm 99.9% sure that the test above doesn't have anything to do with the issue that this revert is trying to fix, because the issue with that revert only happens when `refCount` is set to `true`. So, I'm pretty sure that the test above does its job correctly.

**Comment by josepot** [comment]
> There was an existing test that attempted to test what's tested by the new test

I don't think so. Because the issue that this PR is solving only happens when the source completes synchronously upon subscription. AFAIK the current test:

> 'should not restart due to unsubscriptions if refCount is true when the source has completed

doesn't complete synchronously on its first subscription.
